Tests/SerializationTransformerTests.swift

```diff
@@ -0,0 +1,215 @@
+//
+//  SerializationTransformerTests.swift
+//  Kakapo
+//
+//  Created by Alex Manzella on 27/06/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+import Quick
+import Nimble
+import SwiftyJSON
+@testable import Kakapo
+
+struct UppercaseTransformer<Wrapped: Serializable>: SerializationTransformer {
+    
+    let wrapped: Wrapped
+    
+    func transform(key key: String) -> String {
+        return key.uppercaseString
+    }
+}
+
+struct LowercaseFirstCharacterTransformer<Wrapped: Serializable>: SerializationTransformer {
+    
+    let wrapped: Wrapped
+    
+    func transform(key key: String) -> String {
+        let characters = key.characters
+        let first = String(characters.prefix(1)).lowercaseString
+        let other = String(characters.dropFirst())
+        return first + other
+    }
+}
+
+class SerializationTransformerSpec: QuickSpec {
+    
+    private struct Dog: JSONAPIEntity {
+        let id: String
+        let dogName: String
+    }
+    
+    private struct JUser: JSONAPIEntity {
+        let id: String
+        let userName: String
+        let doggyDog: Dog
+    }
+    
+    struct User: Serializable {
+        let name: String
+    }
+    
+    struct Friend: Serializable {
+        let friends: [User]
+        
+        init(friends: [User]) {
+            self.friends = friends
+        }
+    }
+    
+    struct Snake: Serializable {
+        let theSnakeCamelFriend: String
+    }
+    
+    override func spec() {
+        
+        let user = User(name: "Alex")
+        let friend = Friend(friends: [user])
+        
+        describe("Serialization Transformers") {
+            it("transforms the keys of a Serializable objects") {
+                let serialized = UppercaseTransformer(wrapped: friend).serialize() as! [String: AnyObject]
```

Comment:
Wrap them yes but in real life serialize() can't be called, you return a serializable wrapped into a transformer in the route handler